io/concatenateHDF5.m

```diff
@@ -0,0 +1,157 @@
+% and end of each trial is set in 'trim'. Bad trials are removed (using the
+% plusmaze text file) and if frames were dropped during a good trial, the 
+% previous frame is used to fill the gap. A new text file is produced
+% indicating the start and stop frames for each trial (as well as the gate
+% up/down frames, the direcitons of the trial and time to complete each
+% trial).
+%
+% Example arguments:
+% tifDir = '/Volumes/COHORT9/cohort9-herrin224/mouse5/day20_ego-right';
+% outputDir = '/Users/jmaxey/Documents/MATLAB/PreFrontal';
+% hdf5Name = 'test.hdf5';
+% downsmpFactor = 0.5;
+% plusmazeName = 'mouse5_day20_ego-right.txt';
+% trim = [10,5];
+%
+% 2015-01-27 Jessica Maxey
+
+outputFileName = fullfile(outputDir,[plusmazeName(1:end-4),'_new.txt']);
+outputFile = fopen(outputFileName,'wt+');
+
+[frame_indices,location_info,time] = getTrialIndicies(fullfile(tifDir,plusmazeName));
+startFrames = frame_indices(:,1);
+
+list = dir(fullfile(tifDir,'*.tif'));
+firstName = fullfile(tifDir,list(1).name);
+[rows,cols] = size(imread(firstName));
+numFiles = length(list);
+
+totalFrames = 1;
+
+downsmpRows = floor(rows*downsmpFactor);
+downsmpCols = floor(cols*downsmpFactor);
+chunkSize = [downsmpRows downsmpCols 1];
+dataset = '/Data/Images';
+h5create(fullfile(outputDir,hdf5Name),dataset,[Inf Inf Inf],'ChunkSize',chunkSize,'Datatype','uint16');
+badTrial = 0;
+skipCount = 0;
+testFrames = 1;
+trialCount = 0;
+
+for i=1:length(list)
+    if(i ~= 1)
+        %%% Find if the start frame of the next recording matches one of
+        %%% the start frame indicies
+        idx = find(testFrames == startFrames);
+        if(isempty(idx))
+            %%% Bad trial
+            badTrial = 1;
+            tif_info = imfinfo(fullfile(tifDir,list(i).name));
+            numBadFrames = length(tif_info);
+            testFrames = testFrames + numBadFrames;
+        else
+            %%% Good trial
+            badTrial = 0;
+        end
+    end
+
+    if(badTrial == 0)
+        %%% Add the good trial to the hdf5 file
+        trialCount = trialCount+1;
+        tifName = fullfile(tifDir,list(i).name);
+        tifInfo = imfinfo(tifName);
+        tifFile = Tiff(tifName,'r');
+        oriFrames = length(tifInfo);
+        
+        xmlName = [tifName(1:end-3),'xml'];
+        xmlData = parse_miniscope_xml(xmlName);
+        
+        %%% Determine if frames were dropped by the miniscope during the
+        %%% trial
+        if(str2num(xmlData.dropped_count) ~= 0)
+            droppedFrames = str2num(xmlData.dropped);
+        else
+            numFrames = oriFrames;
+            droppedFrames = 0;
+        end
+        
+        %%% Read in the images, trim the frames at the beginning and end 
+        %%% of the trial (set by the 'trim' argument), and downsample by the 
+        %%% 'downsmpFactor' argument
+        imageStack = uint16(zeros(floor(rows*downsmpFactor),floor(cols*downsmpFactor),oriFrames-trim(1)-trim(2)));

```

Comment:
Should write: `zeros(downsmpRows, downsmpCols, oriFrames-trim(1)-trim(2), 'uint16')`. This avoids allocating a `double` array then converting to `uint16`. Also, you previously defined the floored downsampled dimensions.